src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -65,7 +65,7 @@ public override async IAsyncEnumerable<ArtifactUri<NuGetArtifactType>> GetArtifa
             
             var basePath = NUGET_DEFAULT_CONTENT_ENDPOINT.EnsureTrailingSlash();
             var nameLowercase = purl.Name.ToLowerInvariant();
-            var versionLowercase = purl.Version.ToLowerInvariant();
+            var versionLowercase = NuGetVersion.Parse(purl.Version.ToLowerInvariant());
 
             yield return new ArtifactUri<NuGetArtifactType>(NuGetArtifactType.Nupkg,
                 $"{basePath}{nameLowercase}/{versionLowercase}/{nameLowercase}.{versionLowercase}.nupkg");
```

Comment:
There is a helper method called `GetNupkgUrl` available - should we use it?

```suggestion
                GetNupkgUrl(purl.Name, purl.Version);
```

Reply:
This and the new helper method look good.

There's some duplication between `GetNupkgUrl` and the new `GetNuspecUrl` that could probably be factored out, but this is fine as-is.